File: reducer.py
#!/usr/bin/env python

import sys

current_id = 'init'
me_cnt     = 0
we_cnt     = 0
tot_cnt    = 0

# input comes from STDIN
for line in sys.stdin:
    # parse the input we got from mapper.py
    fields = line[:-1].split('\t')
    if (current_id == fields[0]):
        me,we,tot = fields[3:]
        me_cnt  += int(me)
        we_cnt  += int(we)
        tot_cnt += int(tot)
    else:
        if (current_id!='init'):
            print '%s\t%s\t%s\t%s\t%s\t%s' % (id,pres,yr,me_cnt,we_cnt,tot_cnt)
        id,yr,pres,me,we,tot = fields
        me_cnt     = int(me)
        we_cnt     = int(we)
        tot_cnt    = int(tot)
        current_id = id
# do not forget to output the last id if needed
if current_id != 'init':
    print '%s\t%s\t%s\t%s\t%s\t%s' % (id,pres,yr,me_cnt,we_cnt,tot_cnt)

File: reducer.R
#!/usr/bin/Rscript
# (c) 2015 copyright brad barker - sample code not supported
# all rights reserved may not be copied or published without permission

input <- file("stdin", "r")

id   <- ''
yr   <- ''
pres <- ''
me   <- 0
we   <- 0
tot  <- 0

while(length(line <- readLines(input, n=1, warn=FALSE)) > 0) {
    fields <- unlist(strsplit(line,'\t'))    
    if (id==fields[1]){
       me   <- me  + as.numeric(fields[4])
       we   <- we  + as.numeric(fields[5])
       tot  <- tot + as.numeric(fields[6]) 
    } else {
       if (id!='') {cat(paste(id,yr,pres,me,we,tot,sep="\t"))
                    cat('\n')}
       id   <- fields[1]
       pres <- fields[2]
       yr   <- fields[3]
       me   <- as.numeric(fields[4])
       we   <- as.numeric(fields[5])
       tot  <- as.numeric(fields[6])
    }
}
cat(paste(id,yr,pres,me,we,tot,sep="\t"))
cat('\n')
close(input)
 
